### PR TITLE
Initial Pagerduty Integration

### PR DESCRIPTION
**Pull request by adrain-cb**
## Fixes Issue

Fixes #

### CODE REVIEW

File: internal/client/pagerduty_client.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+			Source:    "Pessimism",
+			Severity:  event.Severity,
+			Timestamp: time.Now(),
+		},
+	}
+}
+
+// marshal ... Marshals the Pagerduty payload
+func (pdp *PagerdutyRequest) marshal() ([]byte, error) {
+	bytes, err := json.Marshal(pdp)
```
Comment (epociask) [inline internal/client/pagerduty_client.go:116]:
```suggestion
func (req *PagerdutyRequest) marshal() ([]byte, error) {
	bytes, err := json.Marshal(req)
```

File: internal/client/pagerduty_client.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	if cfg.IntegrationKey == "" {
+		logging.NoContext().Warn("No Pagerduty integration key provided")
+	}
+
+	return pagerdutyClient{
+		integrationKey:  cfg.IntegrationKey,
+		changeEventsURL: cfg.ChangeEventsURL,
+		alertEventsURL:  cfg.AlertEventsURL,
+		client:          &http.Client{},
+	}
```
Comment (epociask) [inline internal/client/pagerduty_client.go:72]: Is there any reason why we're not returning a pointer?

File: internal/client/pagerduty_client.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+type PagerdutyClient interface {
+	PostEvent(ctx context.Context, event *PagerdutyEventTrigger) (*PagerdutyAPIResponse, error)
+}
+
+type pagerdutyClient struct {
+	integrationKey  string
+	changeEventsURL string
+	alertEventsURL  string
+	client          *http.Client
+}
```
Comment (epociask) [inline internal/client/pagerduty_client.go:59]: Let's add docstring comments

**Review by epociask — COMMENTED** [review]
We should be testing the e2e flow with pagerduty using the [e2e](https://github.com/base-org/pessimism/tree/master/e2e) framework

File: internal/config/config.go [hunk 4]
```diff
@@ -26,13 +27,11 @@ type Config struct {
...
-	SlackChannel string
-
-	SystemConfig  *subsystem.Config
-	ServerConfig  *server.Config
-	MetricsConfig *metrics.Config
+	SystemConfig          *subsystem.Config
+	ServerConfig          *server.Config
+	MetricsConfig         *metrics.Config
+	SlackClientConfig     *client.SlackConfig
+	PagerdutyClientConfig *client.PagerdutyConfig
```
[the remark this answers is not in the document]
Comment (adrain-cb) [reply internal/config/config.go]: [6bb3dd7](https://github.com/base-org/pessimism/pull/130/commits/6bb3dd773305165e38a2d756dbf01f520c37075d)

File: internal/core/constants.go [hunk 5]
```diff
@@ -137,15 +137,18 @@ func StringToHeuristicType(stringType string) HeuristicType {
 type AlertDestination uint8
 
 const (
-	Slack      AlertDestination = iota + 1
-	ThirdParty                  // 2
+	Slack AlertDestination = iota + 1
+	Pagerduty
+	ThirdParty // 2
```
[the remark this answers is not in the document]
Comment (adrain-cb) [reply internal/core/constants.go]: [6bb3dd7](https://github.com/base-org/pessimism/pull/130/commits/6bb3dd773305165e38a2d756dbf01f520c37075d)

File: internal/client/pagerduty_client.go (hunk 1 above)
Comment (adrain-cb) [reply internal/client/pagerduty_client.go]: [6bb3dd7](https://github.com/base-org/pessimism/pull/130/commits/6bb3dd773305165e38a2d756dbf01f520c37075d)

File: internal/client/pagerduty_client.go (hunk 2 above)
Comment (adrain-cb) [reply internal/client/pagerduty_client.go]: [6bb3dd7](https://github.com/base-org/pessimism/pull/130/commits/6bb3dd773305165e38a2d756dbf01f520c37075d)

File: internal/client/pagerduty_client.go (hunk 3 above)
Comment (adrain-cb) [reply internal/client/pagerduty_client.go]: [6bb3dd7](https://github.com/base-org/pessimism/pull/130/commits/6bb3dd773305165e38a2d756dbf01f520c37075d)

**Review by epociask — APPROVED** [review]
LGTM

**Pull request merged**